sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CoalesceShufflePartitions.scala

```diff
@@ -121,6 +122,15 @@ case class CoalesceShufflePartitions(session: SparkSession) extends AQEShuffleRe
     }
   }
 
+  private def advisoryPartitionSize(shuffleStages: Seq[ShuffleStageInfo]): Long = {
```

Comment:
to be safe, shall we only apply this advisory size if there is only one shuffle (which is the case of the final write file stage)?